### PR TITLE
New release of Ada Util, Ada EL, Ada Wiki, ADO, Ada Servlet, ...

### PR DESCRIPTION
**Pull request by stcarrez**
I've made a new release of several projects taking the opportunity to update the crates to the new formats
and so on.

### CODE REVIEW

**Comment by Fabien-Chouteau** [comment]
@mosteo  There is something strange in the CI:
```
WARNING: Deprecated metadata possibly detected at /home/runner/work/alire-index/alire-index/alire/alire.toml

Due to recent changes to Alire's way of storing metadata,
you need to reinitialize or migrate the workspace.
Please check here for details on how to migrate your metadata:

   https://github.com/alire-project/alire/wiki/2020-Metadata-format-migration
```

File: index/ut/utilada_aws/utilada_aws-2.3.0.toml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+licenses = ["Apache 2.0"]
+maintainers = ["Stephane.Carrez@gmail.com"]
+maintainers-logins = ["stcarrez"]
+project-files = [".alire/aws/utilada_aws.gpr"]
+tags = ["web", "http"]
+website = "https://gitlab.com/stcarrez/ada-util"
+
+[[depends-on]]
+utilada = "^2.3.0"
+aws = "any"
```
Comment (Fabien-Chouteau) [inline index/ut/utilada_aws/utilada_aws-2.3.0.toml:15]: @stcarrez were you able to check if this works with the new `aws` source crate in the index?

**Comment by mosteo** [comment]
@stcarrez , the problems with the CI checks are already fixed. If you could rebase your PR into the tip of the branch, the tests will be run properly. Likewise for #200. Thanks!

File: index/ut/utilada/utilada-2.3.0.toml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+[gpr-set-externals."case(os)".linux."case(word_size)".bits-64]
+UTIL_OS = "linux64"
+
+[gpr-set-externals."case(os)".macos]
+UTIL_OS = "macos64"
+[gpr-set-externals."case(os)".windows."case(word_size)".bits-32]
+UTIL_OS = "windows32"
+
+[gpr-set-externals."case(os)".windows."case(word_size)".bits-64]
+UTIL_OS = "windows64"
```
Comment (mosteo) [inline index/ut/utilada/utilada-2.3.0.toml]: According to line 16 in this same file, shouldn't these be `win32` and `win64`?

**Review by mosteo — CHANGES_REQUESTED** [review]

File: index/ut/utilada/utilada-2.3.0.toml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+tags = ["logging", "processes", "streams", "json", "beans", "encoders", "decoders"]
+website = "https://gitlab.com/stcarrez/ada-util"
+
+[gpr-externals]
+BUILD = ["distrib", "debug", "optimize", "profile", "coverage"]
+UTIL_LIBRARY_TYPE = ["relocatable", "static", "static-pic"]
+UTIL_OS = ["win32", "win64", "linux32", "linux64", "macos64", "netbsd32", "netbsd64", "freebsd32", "freebsd64"]
+
+[gpr-set-externals]
+BUILD = "distrib"
```
Comment (Fabien-Chouteau) [inline index/ut/utilada/utilada-2.3.0.toml]: You shouldn't force the users into a fixed `BUILD` or `LIBRARY_TYPE`.

Also using such a common word as `BUILD` for an external is likely to cause name clashes, it is recommended to prefix you externals with the name of the crate/project: `UTILADA_BUILD`.

**Review by Fabien-Chouteau — CHANGES_REQUESTED** [review]

File: index/ut/utilada/utilada-2.3.0.toml (hunk 2 above)
Comment (stcarrez) [reply index/ut/utilada/utilada-2.3.0.toml]: Yes, this is not correct. I've fixed that. Thanks!

File: index/ut/utilada/utilada-2.3.0.toml (hunk 3 above)
Comment (stcarrez) [reply index/ut/utilada/utilada-2.3.0.toml]: Ok, I've removed the BUILD. This is the only place where I've put that and it's not necessary.

File: index/ut/utilada_aws/utilada_aws-2.3.0.toml (hunk 1 above)
Comment (stcarrez) [reply index/ut/utilada_aws/utilada_aws-2.3.0.toml:15]: Yes, it works! Thanks!

**Comment by mosteo** [comment]
I'm experimenting with these crates and I'm hitting a couple of problems.

First, `alr get --build utilada_aws` is not building, with this error:
```
==============Messages for file: /tmp/a/utilada_aws_2.3.0_7fb2e06f/.alire/aws/utilada_aws.gpr
    12.    Has_Aws     : Config.Yes_No := "yes";
                         |
        >>> unknown project "config"

    14.    case Has_Aws is
           1    2
        >>> warning: no when others for this case construction
        >>> unknown variable "Has_Aws"
gprbuild: ".alire/aws/utilada_aws.gpr" processing failed
```

Second, `openapi` seems to somehow depend on `aws` via `utilada_aws`, which is not among the solved dependencies:
```
alr show openapi --graph
Dependencies (direct):
   security^1.3.0
   utilada^2.3.0
   utilada_xml^2.3.0
Dependencies (graph):                                         

                        ┌────────────────────────────────────────────────┐
                        │                                                │
                        │                                                │
  ┌─────────────────────┼──────────────────────┐                         │
  │                     │                      ∨                         ∨
┌───────────────┐     ┌────────────────┐     ┌───────────────────┐     ┌───────────────┐┌───────────────┐
│ openapi=0.3.0 │ ──> │ security=1.3.1 │ ──> │ utilada_xml=2.3.0 │ ──> │ utilada=2.3.0 ││ xmlada=21.0.0 │
└───────────────┘     └────────────────┘     └───────────────────┘     └───────────────┘└───────────────┘
  │                                            │                         ∧                ∧
  └────────────────────────────────────────────┼─────────────────────────┘                │
                                               │                                          │
                                               │                                          │
                                               └──────────────────────────────────────────┘
```

And the error:
```
$ alr get --build openapi
# Previous output omitted here...
==============Messages for file: /tmp/a/openapi_0.3.0_ccfbceed/alire/cache/dependencies/utilada_2.3.0_7fb2e06f/utilada_aws.gpr
     3. with "aws";
             |
        >>> unknown project file: "aws"
        >>> imported by "/tmp/a/openapi_0.3.0_ccfbceed/alire/cache/dependencies/utilada_2.3.0_7fb2e06f/utilada_aws.gpr"
        >>> imported by "/tmp/a/openapi_0.3.0_ccfbceed/alire/cache/dependencies/utilada_2.3.0_7fb2e06f/.alire/utilada_http.gpr"
        >>> imported by "/tmp/a/openapi_0.3.0_ccfbceed/.alire/openapi.gpr"
gprbuild: ".alire/openapi.gpr" processing failed
```

Not sure what's the root of the problems, but at least it seems some dependency is missing somewhere.

**Comment by stcarrez** [comment]
There was an error in the GNAT projects used for Alire on both utilada_aws and openapi.
I've fixed them.

**Comment by Fabien-Chouteau** [comment]
@stcarrez I see that the updates I was looking for are here...

**Comment by jquorning** [comment]
Is it possible to trigger the CI checks again? The logs are not available due to age.

**Comment by mosteo** [comment]
> Is it possible to trigger the CI checks again? The logs are not available due to age.

Curiously, the option to re-run them is also not available. I'll rebase the branch onto 1.0 and push it, that should do it.

**Comment by Fabien-Chouteau** [comment]
@mosteo I don't know why the checks are not running...
I can't re-start them.

**Comment by mosteo** [comment]
I've rebased again and that seems to have fixed the stuck checks.

**Comment by mgrojo** [comment]
How could this be moved forward? I was unable to see the failing build log, so I couldn't get an idea about the problem.

**Comment by mosteo** [comment]
Thanks for pushing me on this. I can't do anything at the moment but I will retrigger the checks ASAP.

**Comment by mgrojo** [comment]
Thanks, Alejandro. Unfortunately, the problem to read the logs hasn't changed. Seems some issue with GitHub, apart from the actual build error.
imagen

**Comment by mosteo** [comment]
@mgrojo , I can see the error logs now, can you?

**Comment by mgrojo** [comment]
Yes, I can see them now.

I think, it is necessary to apply this change to `alire-index/index/ut/utilada/utilada-2.3.0.toml`: https://github.com/alire-project/alire-index/commit/fa6a87fd49df0b71f5f0a8dc5d121b3851bcde3a

**Review by mosteo — DISMISSED** [review]

**Comment by jquorning** [comment]
This version of Ada Util has a bug which is exposed by a recent gnat compiler.

**Comment by mosteo** [comment]
This PR should be reopened against the `stable-1.2` branch, which is the current stable. In my experience it is not enough to rebase and change the base branch, due to some limitation in the Github actions framework, so I'm closing it now. Sorry for the inconvenience.

**Pull request closed without merging**